Backend/src/routers/dashboard.py

```
@@ -16,87 +17,278 @@
-                    select(func.count(func.distinct(TeacherAssignment.subject_id)))
-                    .where(TeacherAssignment.teacher_id == teacher.id)
-                ).first() or 0,
+                "subjects_taught": subjects_taught_res.one(),
             }
+
     return stats
 
 
+@router.get("/recent-activity")
+async def get_recent_activity(
+    current_user: User = Depends(get_current_active_user),
```

Ruff (0.13.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)